packages/eslint-plugin/src/rules/prefer-find.ts

```diff
@@ -0,0 +1,283 @@
+          expression.expressions.at(-1),
+          'Expected to have more than zero expressions in a sequence expression',
+        );
+        return parseIfArrayFilterExpression(lastExpression);
+      }
+
+      if (expression.type === AST_NODE_TYPES.ChainExpression) {
+        return parseIfArrayFilterExpression(expression.expression);
+      }
+
+      // Check if it looks like <<stuff>>(...), but not <<stuff>>?.(...)
+      if (
+        expression.type === AST_NODE_TYPES.CallExpression &&
+        !expression.optional
+      ) {
+        const callee = expression.callee;
+        // Check if it looks like <<stuff>>.filter(...) or <<stuff>>['filter'](...),
+        // or the optional chaining variants.
+        if (callee.type === AST_NODE_TYPES.MemberExpression) {
+          const isBracketSyntaxForFilter = callee.computed;
+          if (
+            isBracketSyntaxForFilter
+              ? getStaticValue(callee.property, globalScope)?.value === 'filter'
+              : callee.property.name === 'filter'
+          ) {
+            const filterNode = callee.property;
+
+            const filteredObjectType = getConstrainedTypeAtLocation(
+              services,
+              callee.object,
+            );
+
+            // As long as the object is an array or an optional chain on a
+            // nullable array, this is an Array.prototype.filter expression.
+            if (isArrayish(filteredObjectType)) {
+              return {
+                isBracketSyntaxForFilter,
+                filterNode,
+              };
+            }
+          }
+        }
+      }
+
+      return undefined;
+    }
+
+    /**
+     * Tells whether the type is a possibly nullable array/tuple or union thereof.
+     */
+    function isArrayish(type: Type): boolean {
+      let isAtLeastOneArrayishComponent = false;
+      for (const unionPart of tsutils.unionTypeParts(type)) {
+        if (
+          tsutils.isIntrinsicNullType(unionPart) ||
+          tsutils.isIntrinsicUndefinedType(unionPart)
+        ) {
+          // ignore
+        } else if (
+          checker.isArrayType(unionPart) ||
+          checker.isTupleType(unionPart)
+        ) {
+          isAtLeastOneArrayishComponent = true;
+        } else {
+          return false;
+        }
+      }
+
+      return isAtLeastOneArrayishComponent;
+    }
+
+    function getObjectIfArrayAtExpression(
+      node: TSESTree.CallExpression,
+    ): TSESTree.Expression | undefined {
+      if (node.arguments.length === 1) {
+        const atArgument = getStaticValue(node.arguments[0], globalScope);
+        if (atArgument != null && isTreatedAsZeroByArrayAt(atArgument.value)) {
+          const callee = node.callee;
+          if (
+            callee.type === AST_NODE_TYPES.MemberExpression &&
+            !callee.optional &&
+            (callee.computed
+              ? getStaticValue(callee.property, globalScope)?.value === 'at'
+              : callee.property.name === 'at')
```

Comment:
[Non-Action] This code here tends to show up a lot: _"check if this call/field/property computes to or is statically a particular value"_. `getStaticValue` works only for the `.computed` side of that. 

Proposal:

* In this PR, can you add a utility in this file?
* Separate from PR, someone should file an issue suggesting using that kind of utility in more places

WDYT?

Reply:
Yep, changed!
And filed https://github.com/typescript-eslint/typescript-eslint/issues/8323

Reply:
Heck yes!